### PR TITLE
(WiP) Upgrade 3.1 for .NET Core 3.1 Compatibility

### PR DESCRIPTION
**Pull request by CrshOverride**
Closes #{ISSUE_NUMBER}

#### BUG FIX
- [ ] reproduce issue in tests
- [ ] fix issue
- [ ] bump package version

#### FEATURE
- [ ] write tests that address the requirements outlined in the issue
- [ ] fulfill the feature requirements
- [ ] bump package version

#### RELATED REPOSITORY UPDATES
- does this feature require documentation? if so, open an issue in the [docs repo](https://github.com/json-api-dotnet/json-api-dotnet.github.io/issues/new)
- does this feature break an API that is implemented in the templates repository? if so, [open an issue](https://github.com/json-api-dotnet/Templates/issues/new)

### CODE REVIEW

File: src/JsonApiDotNetCore/JsonApiDotNetCore.csproj [hunk 1]
```diff
@@ -1,20 +1,20 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <VersionPrefix>3.1.0</VersionPrefix>
+    <VersionPrefix>3.1.1</VersionPrefix>
     <TargetFrameworks>$(NetStandardVersion)</TargetFrameworks>
     <AssemblyName>JsonApiDotNetCore</AssemblyName>
-    <PackageId>JsonApiDotNetCore</PackageId>
+    <PackageId>LegacyJsonApiDotNetCore</PackageId>
```
Comment (bart-degreed) [inline src/JsonApiDotNetCore/JsonApiDotNetCore.csproj:6]: Hi @CrshOverride,

Do you intend to publish and maintain your own version of JsonApiDotNetCore based on v3?

I recommend to upgrade to v4 instead, which contains a ton of fixes and improvements. What's the reason that you want to stick with v3.x instead?

Comment (CrshOverride) [reply src/JsonApiDotNetCore/JsonApiDotNetCore.csproj:6]: @bart-degreed We're currently relying heavily on JSON:API Operations in our application and the breaking changes w/ V4 make it a non-starter. Until we're able to figure out a path forward, we're planning on maintaining a custom fork.

Comment (maurei) [reply src/JsonApiDotNetCore/JsonApiDotNetCore.csproj:6]: I think you'd be happy to hear that the next feature on the roadmap for v4 is support for [atomic operations](https://jsonapi.org/ext/atomic/)

Comment (bart-degreed) [reply src/JsonApiDotNetCore/JsonApiDotNetCore.csproj:6]: @CrshOverride Without making any promises upfront, would you be open to meet with @maurei and me so we can better understand the breaking changes that are blocking your path forward, and maybe explore ways to keep maintaining a single repo? I can organize a Zoom meeting for next week, if you like.

Comment (dashcraft) [reply src/JsonApiDotNetCore/JsonApiDotNetCore.csproj:6]: Same here, going from legacy filters which in reality are the current standard for most if not all other jsonapi spec implementations breaks all the frontend jsonapi interfaces.

I.e. Spraypaint, mobx async store, apollo jsonapi, etc would all break with filtering the way you've currently introduced it.
If we could have an option to enable legacy filters etc, i think that'd be awesome, giving us the ability to migrate over time vs rewriting swaths of frontend interfaces.

Also, v3 of .netcore has been out for years, yet your compatibility recommendation for .netcore v3 is your release candidate v4? That's an odd recommendation. Are there known issues when using .netcore 3 with your current version (v3)?

**Comment by bart-degreed** [comment]
I didn't intend to close this, it happened automatically when I was cleaning up old branches.

**Comment by bart-degreed** [comment]
Closing, as the work here seems to have been abandoned and we don't intend to invest in officially supporting this combination. Having said that, we're willing to work with the community, so please reach out if you want to put effort in this.

**Pull request closed without merging**